src/main/java/com/google/error_analyzer/backend/Pagination.java

```diff
@@ -0,0 +1,233 @@
+/**Copyright 2019 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+package com.google.error_analyzer.backend;
+
+import com.google.gson.Gson;
+import java.util.*;
+import java.lang.*;
+import com.google.error_analyzer.data.SearchErrors;
+import com.google.error_analyzer.data.ErrorFixes;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+
+
+@WebServlet("/pagination")
+public class Pagination extends HttpServlet {
+
+    private String field = "name";
+    private static String ERROR = "errors";
+    private ArrayList < String > data = new ArrayList();
+    private static final Logger LOG = LogManager.getLogger(Pagination.class);
+    public int recordsPerPage = 3;
+    // keep it a odd no so that there are equal pages in front and back 
+    private int noOfPages = 5;
+    private int extraPageInFrontAndBack = noOfPages / 2;
+    private int noOfRecordsOnLastPage = recordsPerPage;
+    private int lastPage = Integer.MAX_VALUE;
```

Comment:
This is a servlet. Same instance of servlet might be used by multiple request (unless specified to dedicate one instance for one request).
For eg. User 1 uploads a file and goes to the paginated view, lastPage is updated for that user. User 2 goes to the paginated view and a different lastPage is used.

User specific variables shouldn't be servlet class variables. Here I saw data and lastPage as user specific.